server/index.js

```diff
@@ -12,10 +15,232 @@ if (process.env.NODE_ENV === 'development') {
 }
 
+app.use(express.json());
+
+function isPositiveInteger(val) {
+  return Number.isInteger(Number(val)) && Number(val) > 0;
+}
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
 app.get('/api/hello', (req, res) => {
   res.json({ hello: 'world' });
 });
 
+// Create a new drawing in the database
+app.post('/api/drawings/', (req, res, next) => {
+  const randomUrlText = urlGenerator(3, 3, 8); // This could be repeated, add in check later
+
+  const sqlCreateNewDrawing = `
+  INSERT INTO "Drawings" ("urlText", "createdByUserId", "dateCreated", "dateSaved", "elements")
+  -- hard code urlText and createdByUserId for now
+  VALUES ($2, $1, NOW(), NULL, '[]')
+  RETURNING "drawingId", "urlText";
+  `;
+  const params = [req.body.userId, randomUrlText];
+
+  db.query(sqlCreateNewDrawing, params)
+    .then(result => {
+      // const { drawingId, urlText, elements } = result.rows[0];
+      res.json(result.rows[0]);
+    })
+    .catch(err => next(err));
+});
+
+// Retrieve the drawn elements for a drawing
+app.get('/api/drawings/:drawingId', (req, res, next) => {
+  const drawingId = req.params.drawingId;
+
+  if (!isPositiveInteger(drawingId)) {
+    res.status(400).json({
+      error: `drawingId must be a positive integer, you supplied: ${drawingId}`
+    });
+    return;
+  }
+
+  const sqlGetDrawing = `
+  SELECT    "drawingId", "elements"
+  FROM      "Drawings"
+  WHERE     "drawingId" = $1;
+  `;
+  const params = [drawingId];
+
+  db.query(sqlGetDrawing, params)
+    .then(result => {
+      const [drawingData] = result.rows;
+      if (!drawingData) {
+        res.status(404).json({
+          error: `cannot find drawing with Id = ${drawingId}`
+        });
+        return;
+      }
+      res.json(drawingData);
+    })
+    .catch(err => next(err));
+});
+
+// Get whether a user has saved a drawing
+app.post('/api/drawingsaves/issaved', (req, res, next) => {
+  if (!req.body || !req.body.userId || !req.body.drawingId) {
+    res.status(400).json({
+      error: 'You did not submit a valid body containing userId and drawingId.'
+    });
+    return;
+  }
+
+  const { userId, drawingId } = req.body;
+
+  if (!isPositiveInteger(userId)) {
+    res.status(400).json({
+      error: `userId must be a positive integer, you supplied: ${userId}`
+    });
+    return;
+  }
+
+  if (!isPositiveInteger(drawingId)) {
+    res.status(400).json({
+      error: `drawingId must be a positive integer, you supplied: ${drawingId}`
+    });
+    return;
+  }
+
+  const sqlCheckIsSaved = `
+  SELECT    1
+  FROM      "DrawingSaves"
+  WHERE     "userId" = $1
+        AND "drawingId" = $2;
+  `;
+  const params = [userId, drawingId];
+
+  db.query(sqlCheckIsSaved, params)
+    .then(result => {
+      (result.rows.length === 0)
+        ? res.status(200).json({
+          isSaved: false
+        })
+        : res.status(200).json({
+          isSaved: true
+        });
```

Comment:
You can use the result of your comparison expression as your `isSaved` value.

```js
res.status(200).json({ isSaved: result.rows.length > 0 });
```

Reply:
💡